storage/remote/queue_manager.go

```diff
@@ -1084,6 +1086,9 @@ func (s *shards) runShard(ctx context.Context, shardID int, queue chan interface
 			// stop reading from the queue. This makes it safe to reference pendingSamples by index.
 			switch d := sample.(type) {
 			case writeSample:
+				if minSampleTs > d.sample.Timestamp {
```

Comment:
Could we not also just check if the samples timestamp is older than the policies min sample age here and cut out even attempting to send that data? That way we'd only hit the policy if samples have been stuck in a retry loop for a while and then become too old.

Reply:
Yes, I think that will be a better idea. We will also introduce a log there, which should be debug, otherwise, it will be too noisy.